### PR TITLE
fix: web-js updates for tp-rework

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

- renamed emailpassword -> emailPassword
- added AllowedDomainsClaim

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `webJsInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/recipe/multitenancy/allowedDomainsClaim.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            hasAccessToCurrentDomain: () => {
+                const includesValidator = this.validators.includes(currentDomain, Number.MAX_SAFE_INTEGER, this.id);
+                return {
+                    id: this.id,
+                    refresh: this.refresh,
+                    shouldRefresh: () => {
+                        return false; // Can't automatically refresh
+                    },
+                    validate: async (payload, userContext) => {
+                        return includesValidator.validate(payload, userContext);
```
Comment (rishabhpoddar) [inline lib/ts/recipe/multitenancy/allowedDomainsClaim.ts:33]:
```suggestion
                        return this.validators.includes(currentDomain, Number.MAX_SAFE_INTEGER, this.id).validate(payload, userContext);
```

Comment (rishabhpoddar) [reply lib/ts/recipe/multitenancy/allowedDomainsClaim.ts:33]: Calculate `currentDomain` inside the validate function itself.

File: lib/ts/recipe/multitenancy/allowedDomainsClaim.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            refresh: async () => {
+                // Can't automatically refresh
+            },
+            defaultMaxAgeInSeconds: Number.MAX_SAFE_INTEGER,
+        });
+
+        // FIXME - maybe it's better if we can add getHost to the interface we are using below, that's in `supertokens-website` npm library
+        const currentDomain = WindowHandlerReference.getReferenceOrThrow()
+            .windowHandler.location.getOrigin()
+            .split("://")[1];
```
Comment (rishabhpoddar) [inline lib/ts/recipe/multitenancy/allowedDomainsClaim.ts:20]: how will this behave for hybrid apps like:
- capacitor
- electron

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/multitenancy/allowedDomainsClaim.ts (hunk 2 above)
Comment (sattvikc) [reply lib/ts/recipe/multitenancy/allowedDomainsClaim.ts:20]: window.location.origin is unavailable in electron app.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**